__tests__/components/editor/ImportFileZone.test.js

```diff
@@ -68,27 +77,31 @@ describe('<ImportFileZone />', () => {
       })
 
       it('displays profile passing validation', async () => {
-        await wrapper.instance().promiseTemplateValidated(template, schemaUrl)
-        expect(wrapper.state().validTemplate).toBeTruthy()
+        template = require('../../__fixtures__/place_profile_no_schema_specified.json')
+        schemaUrl = wrapper.instance().schemaUrl(template)
+        const resolution = await wrapper.instance().promiseTemplateValidated(template, schemaUrl)
+
+        expect(resolution).toBeUndefined()
       })
     })
   })
 
   describe('not schema valid', () => {
     const wrapper = shallow(<ImportFileZone />)
-    let schemaUrl; let
-      template
+    let schemaUrl
+    let template
 
     it('gets the schemaUrl from the resource-template', () => {
       template = require('../../__fixtures__/lcc_v0.2.0_invalid.json')
       schemaUrl = wrapper.instance().schemaUrl(template)
       expect(schemaUrl).toEqual('https://ld4p.github.io/sinopia/schemas/0.2.0/resource-template.json')
     })
 
-    it('displays an error message when missing required property', async () => {
-      await wrapper.instance().promiseTemplateValidated(template, schemaUrl).catch((err) => {
-        expect(wrapper.state().validTemplate).toBeFalsy()
-        expect(err.toString()).toMatch('should have required property')
+    it('displays an error message when missing required property', () => {
+      template = require('../../__fixtures__/lcc_v0.2.0_invalid.json')
+      schemaUrl = wrapper.instance().schemaUrl(template)
+      return wrapper.instance().promiseTemplateValidated(template, schemaUrl).catch((err) => {
+        return expect(err.toString()).toMatch('should have required property')
```

Comment:
I will keep this pattern in mind when testing expected results from promises 👍

Reply:
:beers: